src/main/java/seedu/duke/ui/Ui.java

```diff
@@ -24,6 +28,55 @@ public static void printGreeting() {
         System.out.println("Hello! This is your CEG Future Academic Planner!");
         System.out.println("What can I do for you?");
     }
+
+    // Helper function for printModulePlan()
+    private static void printSemesterTableHeader(String... semesters) {
+        System.out.print("|");
+        for (String semester : semesters) {
+            System.out.printf("%-11s|", semester);
+        }
+        System.out.println();
+    }
+
+    // Helper function for printModulePlan()
+    private static void printModulesForSemesters(
+            Map<Integer, ArrayList<Module>> moduleBySemMap,
+            int startSem,
+            int endSem
+    ) {
+        // Get the max number of modules per sem between the start sem and end sem.
+        int maxModules = 0;
+        for (int i = startSem; i <= endSem; i++) {
+            ArrayList<Module> moduleForSemester = moduleBySemMap.getOrDefault(i, new ArrayList<>());
+            maxModules = Math.max(maxModules, moduleForSemester.size());
+        }
+
+        // Print the module codes and grades under each semester. Prints horizontally.
+        for (int j = 0; j < maxModules; j++) {
+            for (int i = startSem; i <= endSem; i++) {
```

Comment:
Maybe can refactor inner loop into another method to avoid nested loops.

Reply:
Sure, refactoring could make it more readable, but I do not think it is possible to avoid nested loops. Iterating over rows (max no. of modules per sem) and columns (total semesters) to print in table format inherently involves some form of nested iteration. 

If i do pull out the inner loop code into another method, i would need to pass in parameters to the new method such as `startSem`, `endSem`, `moduleBySemMap` as well as the row `index` (which is `j` in this code above). Given the nature of the task is to print out a table, showing this as a nested loop seems appropriate and refactoring might not offer significant benefits in terms of code clarity or maintainability, since it is likely I would not be using the inner loop method outside of this table printing method anyways. Thanks for the comment anyways!